trackbasket_be/models/krogerservice.py

```diff
@@ -66,27 +66,33 @@ def item_search(cls, term, at_risk_user_id):
     nearest_store = user.store[0]
     nearest_store_id = nearest_store.location_id
     parameters = {'filter.term': term, 'filter.locationId': nearest_store_id}
+    if term == '':
+      return 'error'
+
     headers = {'Authorization': 'Bearer {}'.format(Krogerservice.return_token())}
     response = requests.get('https://api.kroger.com/v1/products', params=parameters, headers=headers)
 
-    json_items = response.json()['data']
-
-    items = []
-
-    for item in json_items:
-      data = {}
+    if response.json()['data'] != []:
+      json_items = response.json()['data']
+      items = []
+      for item in json_items:
+        data = {}
+        data['description'] = item['description'] if 'description' in item.keys() else ''
+        data['upc'] = item['upc'] if 'upc' in item.keys() else ''
+        if ('aisleLocations' in item.keys()) and (item['aisleLocations'] != []):
+          data['aisle_number'] = int(item['aisleLocations'][0]['number'])
+        if ('price' in item['items'][0].keys()):
+          data['unit_price'] = item['items'][0]['price']['regular']
```

Comment:
checks the presence of keys in the `response.json()` hash to handle gracefully situations where some attributes are missing from the response (try to search for the item 'oragne' for example)

Reply:
good call on this edge case